### ISSUE TITLE
Stack trace when fetching completions with the internal SDK in --lsp mode

### ISSUE THREAD

**Issue by natebosch** [dart-lang/sdk#39308]
```
NoSuchMethodError: The method 'contains' was called on null.
Receiver: null
Tried calling: contains("expect")
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      CompletionRanking.rerank.<anonymous closure> (package:analysis_server/src/services/completion/dart/completion_ranking.dart:162:58)
#2      List.forEach (dart:core-patch/growable_array.dart:283:8)
#3      CompletionRanking.rerank (package:analysis_server/src/services/completion/dart/completion_ranking.dart:156:13)
<asynchronous suspension>
#4      DartCompletionManager.computeSuggestions (package:analysis_server/src/services/completion/dart/completion_manager.dart:191:35)
<asynchronous suspension>
#5      CompletionHandler._getItems (package:analysis_server/src/lsp/handlers/handler_completion.dart:151:29)
<asynchronous suspension>
#6      CompletionHandler.handle.<anonymous closure> (package:analysis_server/src/lsp/handlers/handler_completion.dart:81:41)
#7      ErrorOr.mapResult (package:analysis_server/lsp_protocol/protocol_speci al.dart:217:12)
#8      CompletionHandler.handle (package:analysis_server/src/lsp/handlers/handler_completion.dart:81:19)
<asynchronous suspension>
#9      MessageHandler.handleMessage (package:analysis_server/src/lsp/handlers/handlers.dart:101:12)
#10     ServerStateMessageHandler.handleMessage (package:analysis_server/src/lsp/handlers/handlers.dart:140:36)
<asynchronous suspension>
#11     LspAnalysisServer.handleMessage.<anonymous closure>.<anonymous closure> (package:analysis_server/src/lsp/lsp_analysis_server.dart:298:49)
<asynchronous suspen sion>
#12     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:258:15)
#13     LspAnalysisServer.handleMessage.<anonymous closure> (package:analysis_server/src/lsp/lsp_analysis_server.dart:293:50)
#14     _rootRun (dart:async/zone.dar t:1124:13)
#15     _CustomZone.run (dart:async/zone.dart:1021:19)
#16     _runZoned (dart:async/zone.dart:1516:10)
#17     runZoned (dart:async/zone.dart:1500:12)
#18     LspAnalysisServer.handleMessage (package:analysis_server/src/lsp/lsp_analysis_server.dart:292:5)
#19     LspByteStreamServerChannel._readMessage.<anonymous closure> (package:analysis_server/src/lsp/channel/lsp_byte_stream_channel.dart:96:18)
#20     _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:258:15)
#21     LspByteStreamServerChann el._readMessage (package:analysis_server/src/lsp/channel/lsp_byte_stream_channel.dart:92:47)
#22     LspByteStreamServerChannel.listen.<anonymous closure> (package:analysis_server/src/lsp/channel/lsp_byte_stream_channel.dart:62:24)
#23     _rootRunUnary (dart:async/zone.dart:11 32:38)
#24     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#25     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#26     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#27     _DelayedData.perform (dart:async/stream_impl.dart:59 1:14)
#28     _StreamImplEvents.handleNext (dart:async/stream_impl.dart:707:11)
#29     _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:667:7)
#30     _rootRun (dart:async/zone.dart:1120:38)
#31     _CustomZone.run (dart:async/zone.dart:1021:19)
#32      _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#33     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#34     _rootRun (dart:async/zone.dart:1124:13)
#35     _CustomZone.run (dart:async/zone.dart:1021:19)
#36     _CustomZone.runGuar ded (dart:async/zone.dart:923:7)
#37     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#38     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#39     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#40     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:116:13)
#41     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:173:5)
```

**Comment by leonsenft** [comment]
Hitting this as well. Let me know if you need any help reproducing.

**Comment by DanTup** [comment]
@lambdabaa this seems to be happening in the completion ranking code. Is this a bug there, or does LSP need to do something to set this up correctly?

**Comment by DanTup** [comment]
@natebosch looks like this was fixed in #39535 by 6f8fa3cbb362f1652dd098b8a35435813cdbadff. I don't have permissions to close this though, so if you're satisfied it is, please close! :-)

**Comment by natebosch** [comment]
Thanks!
